packages/app/src/sections/dashboard/RewardsTab.tsx

```diff
@@ -0,0 +1,382 @@
+import { ZERO_WEI } from '@kwenta/sdk/constants'
+import { formatDollars, formatPercent, truncateNumbers } from '@kwenta/sdk/utils'
+import { wei } from '@synthetixio/wei'
+import { BigNumber } from 'ethers'
+import { formatEther } from 'ethers/lib/utils.js'
+import { FC, useCallback, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import Button from 'components/Button'
+import { FlexDivCol, FlexDivRow, FlexDivRowCentered } from 'components/layout/flex'
+import LabelContainer from 'components/Nav/DropDownLabel'
+import Select, { DropdownIndicator, IndicatorSeparator } from 'components/Select'
+import { Body, Heading } from 'components/Text'
+import { EXTERNAL_LINKS } from 'constants/links'
+import { NO_VALUE } from 'constants/placeholder'
+import useIsL2 from 'hooks/useIsL2'
+import useGetFile from 'queries/files/useGetFile'
+import useGetFuturesFee from 'queries/staking/useGetFuturesFee'
+import useGetFuturesFeeForAccount from 'queries/staking/useGetFuturesFeeForAccount'
+import {
+	FuturesFeeForAccountProps,
+	FuturesFeeProps,
+	TradingRewardProps,
+} from 'queries/staking/utils'
+import { StakingCard } from 'sections/dashboard/Stake/card'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { claimMultipleAllRewards } from 'state/staking/actions'
+import { setSelectedEpoch } from 'state/staking/reducer'
+import {
+	selectEpochData,
+	selectKwentaRewards,
+	selectOpRewards,
+	selectSelectedEpoch,
+	selectSnxOpRewards,
+} from 'state/staking/selectors'
+import { selectNetwork, selectWallet } from 'state/wallet/selectors'
+import media from 'styles/media'
+
+type EpochValue = {
+	period: number
+	start: number
+	end: number
+	label: string
+}
+
+const RewardsTabs: FC<TradingRewardProps> = ({
+	period = 0,
+	start = 0,
+	end = Math.floor(Date.now() / 1000),
+}) => {
+	const { t } = useTranslation()
+	const dispatch = useAppDispatch()
+	const network = useAppSelector(selectNetwork)
+	const isL2 = useIsL2()
+	const epochData = useAppSelector(selectEpochData)
+	const selectedEpoch = useAppSelector(selectSelectedEpoch)
+	const walletAddress = useAppSelector(selectWallet)
+	const kwentaRewards = useAppSelector(selectKwentaRewards)
+	const opRewards = useAppSelector(selectOpRewards)
+	const snxOpRewards = useAppSelector(selectSnxOpRewards)
+
+	const futuresFeeQuery = useGetFuturesFeeForAccount(walletAddress!, start, end)
```

Comment:
Would be good if we refactored this to live in the sdk and fetch from redux the same way as other queries as we'd like to remove the use of react-query eventually.

Reply:
Yes. Good idea. I will refactor this later.